### PR TITLE
Rewrite TypeOfFacilities Tests

### PR DESCRIPTION
**Pull request by LanceGundersen**
## Description
We want to rewrite our Enzyme based tests with tests in RTL that do not test implementation details

## Testing done
Local and unit

## Screenshots
image

## Acceptance criteria
- [ ] Test coverage matches
- [ ] RTL Tests fully replace enzyme tests

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage2.unit.spec.jsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+  renderWithStoreAndRouter,
+  setTypeOfCare,
+} from '../../mocks/setup';
+import { mockFetch, resetFetch } from 'platform/testing/unit/helpers';
+
+const initialState = {
+  featureToggles: {
+    vaOnlineSchedulingVSPAppointmentNew: false,
+    vaOnlineSchedulingDirect: true,
+    // eslint-disable-next-line camelcase
```
[the remark this answers is not in the document]
Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage2.unit.spec.jsx]: File deleted

File: src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx [hunk 2]
```diff
@@ -1,108 +1,108 @@
...
-    };
-
-    const form = mount(
-      <TypeOfFacilityPage
-        openFormPage={openFormPage}
-        updateFormData={updateFormData}
-        history={history}
-        data={{}}
-      />,
+  it('should show form with single required facility question, fill, and submit', async () => {
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx:55]: This reads a little weird to me, maybe something like "should continue to the correct page based on type choice"

File: src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx [hunk 3]
```diff
@@ -1,108 +1,108 @@
...
+    const store = createTestStore(initialState);
+    const screen = renderWithStoreAndRouter(<TypeOfFacilityPage />, {
+      store,
+    });
 
-    expect(form.find('.usa-input-error').length).to.equal(1);
-    expect(history.push.called).to.be.false;
-    form.unmount();
+    fireEvent.click(screen.getByRole('button', { name: /continue »/i }));
+    await expect(screen.getByRole('alert')).to.contain.text(
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx:50]: This await isn't doing anything, I think you want to do `expect(await screen.findByRole(...`

**Review by jbalboni — COMMENTED** [review]
Couple minor things, looks good otherwise

File: src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx (hunk 2 above)
Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx:55]: Yeah that reads nicer 👍

File: src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx (hunk 3 above)
Comment (LanceGundersen) [reply src/applications/vaos/tests/new-appointment/components/TypeOfFacilityPage.unit.spec.jsx:50]: 👍

**Review by jbalboni — APPROVED** [review]

**Pull request merged**